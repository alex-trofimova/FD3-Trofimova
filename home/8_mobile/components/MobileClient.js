import React from 'react';
import PropTypes from 'prop-types';

import {mobileCompanyEvents} from './events';

import './MobileClient.css';

class MobileClient extends React.PureComponent {

  static propTypes = {
    info: PropTypes.shape({
      id: PropTypes.number.isRequired,
      fam: PropTypes.string.isRequired,
      im: PropTypes.string.isRequired,
      otch: PropTypes.string.isRequired,
      balance: PropTypes.number.isRequired,
      status: PropTypes.string.isRequired,
    })
  };

  state = {
    info: this.props.info,
  };

  componentWillReceiveProps = (newProps) => {
    this.setState( {info:newProps.info} );
  };

  delete = () => {
    mobileCompanyEvents.emit('EClientToDelete',this.state.info.id);
  }

  edit = () => {
    mobileCompanyEvents.emit('EClientToEdit',this.state.info);
  }

  render() {

    console.log("MobileClient id="+this.state.info.id+" render");
    
    return (
      <tr className='MobileClient'>
        <td className='ClientCell ClientFam'>{this.state.info.fam}</td>
        <td className='ClientCell ClientIm'>{this.state.info.im}</td>
        <td className='ClientCell ClientOtch'>{this.state.info.otch}</td>
        <td className='ClientCell ClientBalance'>{this.state.info.balance}</td>
        <td className='ClientCell ClientStatus'>{this.state.info.status}</td>
        <td className='ClientCell ClientEdit'>
          <button className='ClientBtn' onClick={this.edit}>редактировать</button>
        </td>
        <td className='ClientCell ClientDelete'>
          <button className='ClientBtn' onClick={this.delete}>удалить</button>
        </td>
      </tr>
      )
  }

}

export default MobileClient;
