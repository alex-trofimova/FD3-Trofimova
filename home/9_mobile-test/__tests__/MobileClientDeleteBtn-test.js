import React from 'react';
import renderer from 'react-test-renderer';

import MobileClient from '../components/MobileClient';
let clientsArr=require('./json/clientsList.json');
let client=clientsArr[1];

test('работа MobileClient DeleteBtn', () => {

// рендер компонента
const component = renderer.create(
  <MobileClient 
  key={client.id} info={client}
  />
);

// проверка утверждений
let componentTree=component.toJSON();
expect(componentTree).toMatchSnapshot();

// найдём в вёрстке компонента саму кнопку
const buttonElem = component.root.find(  el => el.props.className == 'ClientBtn DeleteBtn' );
// и "нажмём" на неё
buttonElem.props.onClick();

// удаление (отмонтирование) компонента
component.unmount();

// проверка утверждений
componentTree=component.toJSON();
expect(componentTree).toMatchSnapshot();

    
});
