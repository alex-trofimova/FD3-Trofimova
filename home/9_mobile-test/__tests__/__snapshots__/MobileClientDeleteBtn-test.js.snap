// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа MobileClient DeleteBtn 1`] = `
<tr
  className="MobileClient"
>
  <td
    className="ClientCell ClientFam"
  >
    Сидоров
  </td>
  <td
    className="ClientCell ClientIm"
  >
    Сидор
  </td>
  <td
    className="ClientCell ClientOtch"
  >
    Сидорович
  </td>
  <td
    className="ClientCell ClientBalance"
  >
    250
  </td>
  <td
    className="ClientCell ClientStatus"
  >
    active
  </td>
  <td
    className="ClientCell ClientEdit"
  >
    <button
      className="ClientBtn EditBtn"
      onClick={[Function]}
    >
      редактировать
    </button>
  </td>
  <td
    className="ClientCell ClientDelete"
  >
    <button
      className="ClientBtn DeleteBtn"
      onClick={[Function]}
    >
      удалить
    </button>
  </td>
</tr>
`;

exports[`работа MobileClient DeleteBtn 2`] = `null`;
